lib/bolt/cli.rb

```diff
@@ -426,58 +426,72 @@ def execute(options)
         return 0
       end
 
-      if options[:mode] == 'plan'
-        executor = Bolt::Executor.new(@config, options[:noop], true)
-        result = pal.run_plan(options[:object], options[:task_options], executor)
-        outputter.print_plan_result(result)
-        # An exception would have been raised if the plan failed
-        code = 0
-      else
-        executor = Bolt::Executor.new(@config, options[:noop])
-        targets = options[:nodes]
-
-        results = nil
-        outputter.print_head
-
-        elapsed_time = Benchmark.realtime do
-          results =
-            case options[:mode]
-            when 'command'
-              executor.run_command(targets, options[:object]) do |event|
-                outputter.print_event(event)
-              end
-            when 'script'
-              script = options[:object]
-              validate_file('script', script)
-              executor.run_script(
-                targets, script, options[:leftovers]
-              ) do |event|
-                outputter.print_event(event)
-              end
-            when 'task'
-              pal.run_task(options[:object],
-                           targets,
-                           options[:task_options],
-                           executor) do |event|
-                outputter.print_event(event)
-              end
-            when 'file'
-              src = options[:object]
-              dest = options[:leftovers].first
+      handler = Signal.trap :INT do |signo|
```

Comment:
This should happen before puppet is loaded probably since that takes so long.

Reply:
ok, I'll need to vary the message depending on when exactly the SIGINT is delivered though